File: 백준/Silver/18258. 큐 2/큐 2.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.NoSuchElementException;
import java.util.StringTokenizer;

public class Main {
	public static void main(String[] args) throws IOException{	   
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int N = Integer.parseInt(br.readLine());
		StringTokenizer st; 
		StringBuilder sb = new StringBuilder(); 
		LinkedList<Integer> q = new LinkedList<Integer>();
		
		for(int i=0; i<N; i++) {
			st = new StringTokenizer(br.readLine()," ");
			switch(st.nextToken()) {
				case "push" : 
					q.offer(Integer.parseInt(st.nextToken()));
					break;
				case "pop" :
					try {
						sb.append(q.remove()).append('\n');
					}catch(NoSuchElementException e) {
						sb.append(-1).append('\n');
					}
				break;
				
				case "size" :
					sb.append(q.size()).append('\n');
					break;
					
				case "empty" :
						if(q.isEmpty()) {
							sb.append(1).append('\n');	
						}else {
							sb.append(0).append('\n');	
						}
						break;
						
				case "front" :
					try {
						sb.append(q.getFirst()).append('\n');	
					}catch(NoSuchElementException e) {
						sb.append(-1).append('\n');	
					}
					break;
					
				case "back" :
					try {
						sb.append(q.getLast()).append('\n');	
					}catch(NoSuchElementException e) {
						sb.append(-1).append('\n');	
					}
					break;
			}
		}
		System.out.println(sb.toString());
    }  
}

File: 백준/Silver/18258. 큐 2/큐 2.py
from collections import deque
from sys import stdin

n = int(stdin.readline())
deque = deque()

for _ in range(n):
    text = stdin.readline().split()
    if text[0] == "push":
        deque.append(int(text[1]))
    elif text[0] == "pop":
        if len(deque) != 0:
            print(deque.popleft())
        else:
            print(-1)
    elif text[0] == "size":
        print(len(deque))
    elif text[0] == "empty":
        if len(deque) != 0:
            print(0)
        else:
            print(1)
    elif text[0] == "front":
        if len(deque) != 0:
            print(deque[0])
        else:
            print(-1)
    else: # back
        if len(deque) != 0:
            print(deque[-1])
        else:
            print(-1)